apisix/plugins/grpc-web.lua

```diff
@@ -123,8 +122,14 @@ function _M.header_filter(conf, ctx)
         core.response.set_header("Access-Control-Allow-Methods", DEFAULT_CORS_ALLOW_METHODS)
         core.response.set_header("Access-Control-Allow-Headers", DEFAULT_CORS_ALLOW_HEADERS)
     end
+
     core.response.set_header("Access-Control-Allow-Origin", DEFAULT_CORS_ALLOW_ORIGIN)
-    core.response.set_header("Content-Type", ctx.grpc_web_mime or DEFAULT_CORS_CONTENT_TYPE)
+
+    local response_mime = ctx.grpc_web_mime
+    if not response_mime then
+        response_mime = core.request.header(ctx, "Content-Type")
+    end
+    core.response.set_header("Content-Type", response_mime)
```

Comment:
We can set the type only when ctx.grpc_web_mime is given?

Reply:
If it is judged that only gRPC-Web type requests can get MIME, I am not sure if this is friendly to other requests.

Reply:
I am confused. The current code just get and reset the Content-Type when the grpc_web_mime is missing.

Reply:
I moved mime type extraction to the plugin's entry point to ensure it is always available in context.